api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -750,22 +750,52 @@ message ScopedRds {
       [(validate.rules).message = {required: true}];
 }
 
+// [#next-free-field: 6]
 message HttpFilter {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.http_connection_manager.v2.HttpFilter";
 
+  // [#not-implemented-hide:] Configuration source specifier for the late-bound
+  // filter configuration. The HTTP Listener is warmed until all the initial
+  // filter configurations are received, unless the flag to apply the default
+  // configuration is set. Subsequent filter updates are atomic, and apply to
+  // new streams while the active streams continue using the older filter
+  // configurations.
+  // If the initial delivery of the filter configuration fails, due to a
+  // timeout for example, the optional default configuration is applied.
+  // Without a default configuration, the filter is disabled, and the HTTP
+  // listener responds with 500 immediately. After the failure, the listener
+  // continues subscribing to the subsequent filter configurations.
+  message HttpFilterConfigSource {
+    config.core.v3.ConfigSource config_source = 1;
+
+    // Optional default configuration to use as the initial configuration if
+    // there is a failure to receive the initial filter configuration or if
+    // `apply_default_config_without_warming` flag is set.
+    google.protobuf.Any default_config = 2;
+
+    // Use the default config as the initial configuration without warming and
+    // waiting for the first xDS response. Requires the default configuration
+    // to be supplied.
+    bool apply_default_config_without_warming = 3;
+  }
+
   reserved 3, 2;
 
   reserved "config";
 
-  // The name of the filter to instantiate. The name must match a
-  // :ref:`supported filter <config_http_filters>`.
+  // The name of the filter configuration. The name is used as a fallback to
+  // select an extension if the type of the configuration proto is not
+  // sufficient. It also serves as a resource name in FCDS.
   string name = 1 [(validate.rules).string = {min_bytes: 1}];
 
   // Filter specific configuration which depends on the filter being instantiated. See the supported
   // filters for further documentation.
   oneof config_type {
     google.protobuf.Any typed_config = 4;
+
+    // [#not-implemented-hide:] Configuration source specifier for FCDS.
+    HttpFilterConfigSource fcds = 5;
```

Comment:
IMHO it's not the best name. `c` for `Config` is not in the name. See LDS, RDS, CDS...
Also, to avoid heavy overload of "filter", could you name it hfds?

Reply:
This discovery service will also serve non-HTTP filters, so HFDS is not appropriate. I named it FDS originally, but renamed it to FCDS per @mattklein123 's suggestion.

Reply:
If we think we will still do filter chain discovery service (FCDS) we can call this FDS. I do wonder if we are reaching a point of acronym failure and should just call it FilterConfigDS or similar.

Reply:
More letters is better IMHO, due to how many xDSes already exist.

Reply:
+1 let's just call it FilterConfigDS unless there are objections. I think we have exhausted the short acronyms.

Reply:
Yep, seems reasonable to me. We should aim to keep these DS names short and snappy still, otherwise people will abbreviate them 😨